### PR TITLE
Resource info (pod name for installer logs while upgrading)

### PR DESCRIPTION
**Pull request by vikramdevtron**
# Description

Generic resource info api added on k8s rest handler.
- pod name for installer logs while upgrading

Fixes # (issue) : https://github.com/devtron-labs/devtron/issues/1549

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

- [x] Resource info fetch api

# Checklist:

* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] I have performed a self-review of my own code
* [ ] I have commented my code, particularly in hard-to-understand areas
* [ ] I have tested it for all user roles
* [ ] I have added all the required unit/api test cases

### CODE REVIEW

File: internal/util/K8sUtil.go [hunk 1]
```diff
@@ -441,3 +442,22 @@ func (impl K8sUtil) GetClientByToken(serverUrl string, token map[string]string)
...
+	if err != nil {
+		impl.logger.Errorw("cluster config error", "err", err)
+		return nil, err
+	}
+	pods, err := client.Pods(namespace).List(metav1.ListOptions{
+		LabelSelector: labelSelector,
+	})
+	if err != nil {
+		return nil, err
+	} else if len(pods.Items) > 1 {
```
Comment (manish-agrawal-ai) [inline internal/util/K8sUtil.go:457]: if length == 0

File: specs/generic.yaml [hunk 2]
```diff
@@ -0,0 +1,35 @@
+openapi: "3.0.2"
+info:
+  title: version api
+  version: "1.0"
+paths:
+  /orchestrator/k8s/resource/info:
```
Comment (manish-agrawal-ai) [inline specs/generic.yaml:6]: this API is not generic. (you are sending hardcoded inception pod name in this API)

File: util/k8s/k8sApplicationService.go [hunk 3]
```diff
@@ -52,6 +56,12 @@ type ResourceRequestBean struct {
 	K8sRequest    *application.K8sRequestBean `json:"k8sRequest"`
 }
 
+const DevtronDefaultNamespace = "devtroncd"
```
Comment (manish-agrawal-ai) [inline util/k8s/k8sApplicationService.go:59]: "devtroncd" can be used from some other place (I assume this is being used somewhere)

File: specs/generic.yaml (hunk 2 above)
Comment (vikramdevtron) [reply specs/generic.yaml:6]: file name is generic only, we can renamed but it contains apis which are used for generic purpose inception pod name is not hardcoded, its one of the generic api from this file. response is based on label selector.

**Review by manish-agrawal-ai — APPROVED** [review]

**Pull request merged**